### ISSUE TITLE
Add position parameter to mobileInjectEmulatorCameraImage

### ISSUE THREAD

**Issue by hadalin** [appium/appium-android-driver#940]
The [`mobileInjectEmulatorCameraImage`](https://github.com/appium/appium-android-driver/pull/931) is very useful, thank you.

Would it be possible to add `position` parameter, along with the [`payload`](https://github.com/appium/appium-android-driver/pull/931/commits/97ae9c046401aaa95dccc0d93779b9b267f837a9#diff-25ac2cf50dd463e37f57f6f603e479c5805b0d7afe8ca809271a955ee3510f4dR1057-R1060) that would allow setting the position of the injected image?

**Comment by mykola-mokhnach** [comment]
It is possible to add such argument, however I don't think it is going to work seamlessly. You'd need to restart the whole emulator in order for it to be picked up. The reason for that is that Google provides no way to change virtual scene params over the telnet interface except of for images themselves (I mean the `virtualscene-image` utility command).

Other settings need to be loaded from the config file directly, which requires the media service to be restarted. And since it is not possible to do that without having the root access, the only way would be to restart the emulator completely.

**Comment by hadalin** [comment]
If I understand correctly, [this config](https://github.com/appium/appium-android-driver/pull/931/commits/97ae9c046401aaa95dccc0d93779b9b267f837a9#diff-e24de1867645419fb2ca024667ad05f3097538428d349419e05794574ec6be4bR9-R19) gets passed to the [virtualscene-image](https://github.com/appium/appium-android-driver/pull/931/commits/97ae9c046401aaa95dccc0d93779b9b267f837a9#diff-e24de1867645419fb2ca024667ad05f3097538428d349419e05794574ec6be4bR52) command. Perhaps I'm missing something, but wouldn't changing `position` affect the position of the injected image without the need to change virtual scene params?

**Comment by hadalin** [comment]
I think I get it. You'd need to restart the emulator for changes to [`CONFIG`](https://github.com/appium/appium-android-driver/pull/931/commits/97ae9c046401aaa95dccc0d93779b9b267f837a9#diff-e24de1867645419fb2ca024667ad05f3097538428d349419e05794574ec6be4bR9-R19) to be applied.

**Comment by mykola-mokhnach** [comment]
> I think I get it. You'd need to restart the emulator for [`CONFIG`](https://github.com/appium/appium-android-driver/pull/931/commits/97ae9c046401aaa95dccc0d93779b9b267f837a9#diff-e24de1867645419fb2ca024667ad05f3097538428d349419e05794574ec6be4bR9-R19) changes to be applied.

Right. The whole emulator must be restarted in order for virtual scene to pick up the changes

**Comment by hadalin** [comment]
Also, when I used `mobileInjectEmulatorCameraImage` for the first time to inject time image (in `appium-uiautomator2-driver`), the image didn't appear in the virtual scene. It appeared later on. I'm not sure when, but probably after I restarted the emulator. Could that be the case?

**Comment by mykola-mokhnach** [comment]
yes, it could be the case. https://github.com/appium/appium-uiautomator2-driver?tab=readme-ov-file#mobile-injectemulatorcameraimage mentions that camera service needs to be restarted after the config is changed for the very first time, although it is probably not very obvious. I'll update it to be clear about emulator restart.

**Comment by mykola-mokhnach** [comment]
@hadalin Maybe try https://github.com/appium/appium-uiautomator2-driver/pull/798 after it is merged. It still needs to restart the emulator, but it's probably ok, since the action is anyway automated

**Comment by hadalin** [comment]
What could be done, though, is to split the functionality into two methods. The first one would be something like `mobileSetupEmulatorCameraImage` that would require a restart. It could also take `size` and `position` parameters. The second one would be `mobileInjectEmulatorCameraImage` as it is. It would possibly be more obvious there's a "setup" phase involved here.

**Comment by mykola-mokhnach** [comment]
I assume it would be painful to do emulator restart mid-session. the reason for it is that the uia2 server must be running there. By restarting the emulator we have to kill it and then restore the session seamlessly. Currently it is not possible without major architecture changes in the server package.
